pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -301,6 +297,21 @@ func (c *InstallerController) manageInstallationPods(operatorSpec *operatorv1.St
 
 		// if we are in a transition, check to see whether our installer pod completed
 		if currNodeState.TargetRevision > currNodeState.CurrentRevision {
+
+			// get current revision status
+			_, err := c.configMapsGetter.ConfigMaps(c.targetNamespace).Get(fmt.Sprintf("revision-status-%d", currNodeState.TargetRevision), metav1.GetOptions{})
+			if apierrors.IsNotFound(err) && operatorStatus.LatestAvailableRevision > currNodeState.CurrentRevision {
+				// current revision status was deleted, set target to latest available revision
+				newCurrNodeState := currNodeState.DeepCopy()
+				newCurrNodeState.TargetRevision = operatorStatus.LatestAvailableRevision
+				if b, err, done := c.updateNodeStatus(newCurrNodeState, currNodeState, fmt.Sprintf("target revision %d no longer available", currNodeState.TargetRevision)); done {
```

Comment:
what exactly is the problem that we need this step? That `ensureInstallerPod` fails if there is no installer pod of the target revision? Wouldn't it be more clever to handle that in `newNodeStateForInstallInProgress`? Depending on some configmap of another controller feels like bandaid.